backend/routes/knowledge.py

```
@@ -0,0 +1,736 @@
+            where_clause["content_type"] = content_type
+        if source_filter:
+            where_clause["source"] = {"$contains": source_filter}
+
+        # Get documents using proper encapsulated method
+        try:
+            result_docs = retriever.semantic_searcher.get_documents(
+                where=where_clause if where_clause else None, limit=limit, offset=offset
+            )
+        except Exception as e:
+            logger.error(f"Failed to get documents: {e}")
+            raise HTTPException(status_code=503, detail="Failed to retrieve documents")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)